### PR TITLE
Upgrade go-restful to latest one

### PR DESCRIPTION
**Pull request by aljesusg**
update k8s client-go

	k8s.io/api v0.27.2
	k8s.io/apimachinery v0.27.2
	k8s.io/client-go v0.27.2

This updated the requirement go-restful library to 3.9

### CODE REVIEW

**Review by jmazzitelli — APPROVED** [review]
1. I did a build and ran unit tests myself. All pass.
```
make build build-ui test
```

2. After building this PR, I pushed the new container to a minikube cluster, and then I deployed the operator and server and tested. All works (Kiali was in anonymous mode when doing this).
```
make -e CLUSTER_TYPE=minikube cluster-push operator-create kiali-create

./hack/kiali-port-forward.sh
```

3. After undeploying everything (`make -e CLUSTER_TYPE=minikube operator-delete`), I then ran the openid molecule test along with a couple other auth related tests on the minikube cluster. All pass.
```
./hack/run-molecule-tests.sh -dorp podman -ce $(which kubectl) -ct minikube -udi true -at "openid-test token-test header-auth-test"

...
=====================
=== TEST RESULTS: ===
=====================

                             openid-test... success [1m 52s]
                              token-test... success [2m 9s]
                        header-auth-test... success [1m 54s]
```

4. I then re-installed kiali and then bookinfo demo and the graph looks good.

File: tests/integration/utils/kube/kube_client.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func DeleteKialiPod(ctx context.Context, kubeClient kubernetes.Interface, namesp
 // Waits for old kiali pod to terminate and for the new one to be ready
 func RestartKialiPod(ctx context.Context, kubeClient kubernetes.Interface, namespace string, currentKialiPod string) error {
 
-	return wait.PollImmediate(time.Second*5, time.Minute*4, func() (bool, error) {
+	return wait.PollUntilContextTimeout(ctx, time.Second*5, time.Minute*4, false, func(ctx context.Context) (bool, error) {
```
Comment (nrfox) [inline tests/integration/utils/kube/kube_client.go:69]:
```suggestion
	return wait.PollUntilContextTimeout(ctx, time.Second*5, time.Minute*4, true, func(ctx context.Context) (bool, error) {
```

File: tests/integration/utils/kube/kube_client.go [hunk 2]
```diff
@@ -146,7 +146,7 @@ func UpdateKialiCR(ctx context.Context, dynamicClient dynamic.Interface, kubeCli
 
 	// Need to know when the kiali operator has seen the CR change and finished updating
 	// the configmap. There's no ObservedGeneration on the Kiali CR so just checking the configmap itself.
-	require.NoError(wait.PollImmediate(time.Second*5, time.Minute*2, func() (bool, error) {
+	require.NoError(wait.PollUntilContextTimeout(ctx, time.Second*5, time.Minute*2, false, func(ctx context.Context) (bool, error) {
```
Comment (nrfox) [inline tests/integration/utils/kube/kube_client.go:149]:
```suggestion
	require.NoError(wait.PollUntilContextTimeout(ctx, time.Second*5, time.Minute*2, true, func(ctx context.Context) (bool, error) {
```

File: kubernetes/cache/proxy_status.go [hunk 3]
```diff
@@ -38,7 +38,7 @@ func (c *kialiCacheImpl) pollIstiodForProxyStatus(ctx context.Context) {
 					)
 
 					interval := c.tokenNamespaceDuration / 2
-					retryErr := wait.PollImmediateUntilWithContext(ctx, interval, func(ctx context.Context) (bool, error) {
+					retryErr := wait.PollUntilContextCancel(ctx, interval, false, func(ctx context.Context) (bool, error) {
```
Comment (nrfox) [inline kubernetes/cache/proxy_status.go:41]:
```suggestion
					retryErr := wait.PollUntilContextCancel(ctx, interval, true, func(ctx context.Context) (bool, error) {
```

nit: looks like this should pass `true` to keep the same behavior as `immediate` where the polling function starts immediately.

**Review by nrfox — APPROVED** [review]
One minor suggestion but not blocking

**Comment by aljesusg** [comment]
@nrfox @jmazzitelli can you approved this one?

**Review by jmazzitelli — APPROVED** [review]

**Comment by aljesusg** [comment]
Thank you @jmazzitelli !!!

File: kubernetes/cache/kube_cache.go [hunk 4]
```diff
@@ -465,12 +504,18 @@ func (c *kubeCache) createGatewayInformers(namespace string) gateway.SharedInfor
...
+			_, err := sharedInformers.Gateway().V1beta1().Gateways().Informer().AddEventHandler(c.registryRefreshHandler)
+			if err != nil {
+				log.Errorf("Error adding Handler to Informer Gateways: %s", err.Error())
+			}
 		}
 		if c.CheckIstioResource(kubernetes.K8sHTTPRoutes) {
 			lister.k8shttprouteLister = sharedInformers.Gateway().V1beta1().HTTPRoutes().Lister()
 			lister.cachesSynced = append(lister.cachesSynced, sharedInformers.Gateway().V1beta1().HTTPRoutes().Informer().HasSynced)
-			sharedInformers.Gateway().V1beta1().Gateways().Informer().AddEventHandler(c.registryRefreshHandler)
+			_, err := sharedInformers.Gateway().V1beta1().Gateways().Informer().AddEventHandler(c.registryRefreshHandler)
```
Comment (jmazzitelli) [inline kubernetes/cache/kube_cache.go:515]: I think this needs to be HTTPRoutes - looks like a copy-paste error?

**Pull request merged**